process/smartContract/process.go

```diff
@@ -1327,35 +1342,54 @@ func (sc *scProcessor) createSmartContractResults(
 
 // createSCRForSender(vmOutput, tx, txHash, acntSnd)
 // give back the user the unused gas money
-func (sc *scProcessor) createSCRForSender(
+func (sc *scProcessor) createSCRForSenderAndRelayer(
 	vmOutput *vmcommon.VMOutput,
 	tx data.TransactionHandler,
 	txHash []byte,
 	callType vmcommon.CallType,
-) *smartContractResult.SmartContractResult {
+) (*smartContractResult.SmartContractResult, *smartContractResult.SmartContractResult) {
 	if vmOutput.GasRefund == nil {
 		vmOutput.GasRefund = big.NewInt(0)
 	}
 
 	storageFreeRefund := big.NewInt(0).Mul(vmOutput.GasRefund, big.NewInt(0).SetUint64(sc.economicsFee.MinGasPrice()))
-	refundErd := core.SafeMul(vmOutput.GasRemaining, tx.GetGasPrice())
+	gasRefund := core.SafeMul(vmOutput.GasRemaining, tx.GetGasPrice())
 
 	rcvAddress := tx.GetSndAddr()
 	if callType == vmcommon.AsynchronousCallBack {
 		rcvAddress = tx.GetRcvAddr()
 	}
 
+	gasRemaining := vmOutput.GasRemaining
+	var refundGasToRelayerSCR *smartContractResult.SmartContractResult
+	relayedSCR, isRelayed := isRelayedTx(tx)
+	if isRelayed && callType != vmcommon.AsynchronousCall && gasRefund.Cmp(zero) > 0 {
```

Comment:
You can extract into a variable the whole condition. Ex.: shouldRefundGasToRelayerSCR

Reply:
done